### PR TITLE
Add ksp plugin

### PR DESCRIPTION
**Pull request by MuXiu1997**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -44,13 +44,14 @@ plugins {
     id 'eclipse'
     id 'scala'
     id 'maven-publish'
-    id 'org.jetbrains.kotlin.jvm'        version '1.5.30' apply false
-    id 'org.jetbrains.kotlin.kapt'       version '1.5.30' apply false
+    id 'org.jetbrains.kotlin.jvm'        version '1.5.30'       apply false
+    id 'org.jetbrains.kotlin.kapt'       version '1.5.30'       apply false
+    id 'com.google.devtools.ksp'         version '1.5.30-1.0.0' apply false
```
Comment (mitchej123) [inline build.gradle:49]: What's the purpose of this plugin?  

Also we don't edit the build scripts on individual mods as they get updated periodically from https://github.com/GTNewHorizons/ExampleMod1.7.10  If it's useful it should be added there and then updated here.

**Review by mitchej123 — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (mitchej123) [reply build.gradle:49]: Should also bump the version at the top with the unix timestamp (can grab from here https://www.unixtimestamp.com/)

Comment (MuXiu1997) [reply build.gradle:49]: > What's the purpose of this plugin?

tldr: like `kapt`, annotation processors


`Kotlin Symbol Processing (KSP) is an API that you can use to develop lightweight compiler plugins. KSP provides a simplified compiler plugin API that leverages the power of Kotlin while keeping the learning curve at a minimum. Compared to [kapt](https://kotlinlang.org/docs/kapt.html), annotation processors that use KSP can run up to 2 times faster.`



> Should also bump the version at the top with the unix timestamp (can grab from here https://www.unixtimestamp.com/)

It looks like the [workflow](https://github.com/GTNewHorizons/ExampleMod1.7.10/blob/master/.github/workflows/buildscript-maintenance.yml) will handle this automatically

Comment (mitchej123) [reply build.gradle:49]: > It looks like the [workflow](https://github.com/GTNewHorizons/ExampleMod1.7.10/blob/master/.github/workflows/buildscript-maintenance.yml) will handle this automatically

It should but it currently fails due to it being a protected branch, and we haven't gotten around to devising a workaround :-/

Comment (MuXiu1997) [reply build.gradle:49]: how about using a workflow to automatically submit a pr for a version update and automatically merge it

Comment (MuXiu1997) [reply build.gradle:49]: [Create Pull Request · Actions · GitHub Marketplace](https://github.com/marketplace/actions/create-pull-request) + [Enable Pull Request Automerge · Actions · GitHub Marketplace](https://github.com/marketplace/actions/enable-pull-request-automerge)

**Review by mitchej123 — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (MuXiu1997) [reply build.gradle:49]: In fact I'm planning to make a workflow to automatically update all other repository buildscript, I already have a solution but I don't have much time these days

Comment (mitchej123) [reply build.gradle:49]: Not against it, just haven't had a chance.  Would also need to be sure the actions didn't run against that merged PR.

Comment (MuXiu1997) [reply build.gradle:49]: If workflow works, there is no need to use gradle to update buildscript, or just update from its own repository, and naturally there is no conflict
